### ISSUE TITLE
Feature request: split tunnel on linux

### ISSUE THREAD

**Issue by pyrrhichios** [Windscribe/Desktop-App#100]
Hi,

I'd like to request you add split tunnelling to the GUI app / the CLI tools for Linux please. This is available with PIA VPN so it should be technically possible.

**Comment by bernerdad** [comment]
Hi.  We have a ticket on the board to investigate the feasibility and time requirements for this on our current milestone (app version 2.8).  There's a high probability we will implement this in some fashion, but likely not until the v2.9 app milestone.

**Comment by pyrrhichios** [comment]
Thanks for the update - great to know it's on the roadmap even if there's no definite date for implementation at this time. Thanks again, I appreciate it!

**Comment by windscribe-eva-01** [comment]
PIA uses additional network cgroup with additional routing table for the apps which need to bypass VPN, and adds the applications automatically by monitoring newly started processes.

In the meantime, you can configure exactly the same manually:
https://people.math.wisc.edu/~emeitner/bypassing-a-vpn-using-cgroups.html

Such configuration won't work with flatpak applications reliably. We'll try to implement it better than PIA in our client 😄

**Comment by bernerdad** [comment]
We have implemented IP/hostname split tunneling in a new guinea pig build.  App-based split tunneling is postponed to the 2.9 release.  We'd very much appreciate any feedback you may have on this initial implementation.  Cheers!
https://deploy.totallyacdn.com/desktop-apps/2.8.1/Windscribe_2.8.1_guinea_pig.exe

**Comment by pyrrhichios** [comment]
Thanks, appreciate the update! Think I'm hitting the limits of my knowledge here - what would IP/hostname split tunnelling look like? Is that just (for example) "any traffic heading to a domain like google.com, ebay.co.uk etc. > send through the tunnel?

I'm a bit short on time to tinker at the weekends right now, but happy to have a play when I can :)

**Comment by bernerdad** [comment]
Yes, IP/hostname spit tunneling allows you to control how apps on your computer access a specific IP address or hostname, like google.com.  In inclusive mode, any attempt to access a specified IP/hostname will be directed over the VPN tunnel.  In exclusive mode, any attempt to access a specified IP/hostname will bypass the VPN tunnel.

**Comment by paul38752** [comment]
how far is 2.9 in development

**Comment by bernerdad** [comment]
> how far is 2.9 in development

We likely won't start development on 2.9 until mid-December.

**Comment by bernerdad** [comment]
Hi.  The per-app split tunneling feature for Linux is now available in the v2.9.2 guinea pig build.

**Comment by pyrrhichios** [comment]
> Hi. The per-app split tunneling feature for Linux is now available in the v2.9.2 guinea pig build.

Brilliant! I'll have a play at the weekend if there's time. Thanks for the update 🙂

**Comment by pyrrhichios** [comment]
> Hi. The per-app split tunneling feature for Linux is now available in the v2.9.2 guinea pig build.

Just tested this out on an old Linux laptop - very easy to set up and works exactly as expected, and confirmed with IP lookup. Thank you! Exactly what I wanted, much appreciated 🙂

**Comment by bernerdad** [comment]
We're very happy this new feature is working as expected for you.  Cheers!
